lib/streamlit/connections/snowflake_connection.py

```diff
@@ -321,28 +301,10 @@ def query(
         >>> st.dataframe(df)
 
         """
-        from snowflake.connector.errors import ProgrammingError  # type: ignore[import]
-        from snowflake.connector.network import (  # type: ignore[import]
-            BAD_REQUEST_GS_CODE,
-            ID_TOKEN_EXPIRED_GS_CODE,
-            MASTER_TOKEN_EXPIRED_GS_CODE,
-            MASTER_TOKEN_INVALD_GS_CODE,
-            MASTER_TOKEN_NOTFOUND_GS_CODE,
-            SESSION_EXPIRED_GS_CODE,
-        )
         from tenacity import retry, retry_if_exception, stop_after_attempt, wait_fixed
 
-        retryable_error_codes = {
-            int(code)
-            for code in (
-                ID_TOKEN_EXPIRED_GS_CODE,
-                SESSION_EXPIRED_GS_CODE,
-                MASTER_TOKEN_NOTFOUND_GS_CODE,
-                MASTER_TOKEN_EXPIRED_GS_CODE,
-                MASTER_TOKEN_INVALD_GS_CODE,
-                BAD_REQUEST_GS_CODE,
-            )
-        }
+        # the ANSI-compliant SQL code for "connection was not established" (see docs: https://docs.snowflake.com/en/developer-guide/python-connector/python-connector-api#id6)
+        SQLSTATE_CONNECTION_WAS_NOT_ESTABLISHED = "08001"
```

Comment:
This seems to be the sqlstate which is set by the snowpark library: https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/network.py#L584-L596
Since it is an official sql standard, it is better to check for that than for the error ids as we did before.